File: 2.py
# Find the sum of all the even-valued terms in the Fibonacci sequence which do not exceed four million.

# need a dynamic programming approach

def fib(n):
    """
    
    Arguments:
    - `n`:
    """
    if (n == 0 or n == 1):
        return 1
    else:
        return fib(n-1) + fib(n-2)
 
print fib(10)
        
    
    
    
    

File: 2.hs
module Main where

fib 0 = 1
fib 1 = 1
fib n = fib(n-1) + fib(n-2)

-- in general would be
s = sum $ filter even $ takeWhile (< 40000) [fib n | n <- [0..]]

main :: IO ()
main = putStrLn $ show s